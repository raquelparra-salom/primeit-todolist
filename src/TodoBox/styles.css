.box {
  margin: auto;
  width: 650px;
  margin-top: 150px;
  background-color: #ffffff;
  border: 2px solid #ffffff;
  border-radius: 12px;
  padding: 5px;
}

.right-button {
  float: right;
}
