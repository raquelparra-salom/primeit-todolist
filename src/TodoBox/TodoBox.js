import { Button, IconButton } from '@mui/material';
import React, { useState } from 'react';
import Header from '../Header';
import TodoList from '../TodoList/TodoList';
import HistoryIcon from '@mui/icons-material/History';
import FormItem from '../FormItem';
import './styles.css';

function TodoBox() {
    const [items, setItems] = useState([])
    const [itemSelected, setItemSelected] = useState(null)
    const [open, setOpen] = useState(false);
    const [itemForm, setItemForm] = useState("")

    const handleChangeItem = (e) => {
        setItemForm(e.target.value);
    }

    const handleClickClose = () => {
        setOpen(false);
        setItemForm("")
    };

    const handleClickOpen = () => {
        setOpen(true);
    }

    const handleClickSelected = (item) => {
        setItemSelected(item)
    }

    const handleDoubleClickRemove = (item) => {
        const index = items.indexOf(item);
        setItems(currentItem => currentItem.filter((item, i) => i !== index));
    }


    const handleClickAdd = () => {
        // let item = prompt("Add item to list");
        if (itemForm !== null && itemForm !== "") {
            setItems([...items, itemForm]);
            setItemForm("")
        }
        setOpen(false);
    }

    const handleClickDelete = () => {
        const index = items.indexOf(itemSelected);
        setItems(currentItem => currentItem.filter((item, i) => i !== index));
        setItemSelected(null)
    }

    const handleClickDeshacer = () => {
        console.log("clicando boton deshacer")
    }


    return (
        <div className="box">
            <Header />
            <TodoList data={items} itemSelected={itemSelected} handleDoubleClickRemove={handleDoubleClickRemove} handleClickSelected={handleClickSelected} />
            <div>
                <IconButton disabled={true} color="primary" aria-label="desahcer cambios" onClick={handleClickDeshacer}>
                    <HistoryIcon />
                </IconButton>
                <div className="right-button">
                    <Button disabled={itemSelected === null} variant="outlined" onClick={handleClickDelete}>Delete</Button>
                    <Button className="margin-button" variant="contained" onClick={handleClickOpen}>Add</Button>
                </div>
            </div>
            <FormItem open={open} handleClickClose={handleClickClose} handleClickAdd={handleClickAdd} handleChangeItem={handleChangeItem} item={itemForm} />

        </div >
    );
}

export default TodoBox;
