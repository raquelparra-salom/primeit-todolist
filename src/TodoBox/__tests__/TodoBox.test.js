import React from 'react';
import { render } from '@testing-library/react';
import TodoBox from '../TodoBox';

test('render TodoBox', () => {
    const { container } = render(
        <TodoBox />);
    expect(container).toMatchSnapshot();
});
