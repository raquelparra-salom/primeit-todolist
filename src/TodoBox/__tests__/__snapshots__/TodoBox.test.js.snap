// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TodoBox 1`] = `
<div>
  <div
    class="box"
  >
    <header>
      <h1>
        This is a technical proof
      </h1>
      <p>
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
      </p>
    </header>
    <div
      class="default-message"
    >
       No hay elementos 
    </div>
    <div>
      <button
        aria-label="desahcer cambios"
        class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HistoryIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
          />
        </svg>
      </button>
      <div
        class="right-button"
      >
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Delete
        </button>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root margin-button css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Add
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div />
  </div>
</div>
`;
